backend/src/src-search/src/main/java/org/eclipse/sw360/search/SearchHandler.java

```diff
@@ -62,19 +71,82 @@ public List<SearchResult> searchFiltered(String text, User user, List<String> ty
         }
 
         // Query user and other database
-        List<SearchResult> results = Lists.newArrayList();
+        Set<SearchResult> results = Sets.newLinkedHashSet();
         if (typeMask.isEmpty() || typeMask.contains(SW360Constants.TYPE_USER)) {
-            results.addAll(dbSw360users.search(text, Arrays.asList(SW360Constants.TYPE_USER), user));
+            if (text.contains("pkg:")) {
+                dealWithSpecialCharacters(text, user, results, dbSw360users, Arrays.asList(SW360Constants.TYPE_USER));
+            } else {
+                results.addAll(dbSw360users.search(text, Arrays.asList(SW360Constants.TYPE_USER), user));
+            }
         }
         if(typeMask.isEmpty() || !typeMask.get(0).equals(SW360Constants.TYPE_USER) || typeMask.size() > 1) {
-            results.addAll(dbSw360db.search(text, typeMask, user));
+            if (text.contains("pkg:")) {
+                dealWithSpecialCharacters(text, user, results, dbSw360db, typeMask);
+            } else {
+                results.addAll(dbSw360db.search(text, typeMask, user));
+            }
         }
-        Collections.sort(results, new SearchResultComparator());
+
+        List<SearchResult> srs = new ArrayList<SearchResult>(results);
+        Collections.sort(srs, new SearchResultComparator());
 
         if (log.isTraceEnabled())
             log.trace("Search for " + text + " returned " + results.size() + " results");
 
-        return results;
+        return srs;
+    }
+
+    private void dealWithSpecialCharacters(String text, User user, Set<SearchResult> results, AbstractDatabaseSearchHandler dbSearchHandler, final List<String> typeMask) {
```

Comment:
Please re-use the code  y making another generic method.

Reply:
Done